core/domain/opportunity_services.py

```diff
@@ -407,6 +407,199 @@ def update_opportunities_with_new_topic_name(topic_id, topic_name):
+    SkillOpportunityModel objects.
+
+    Args:
+        skill_opportunities: list(SkillOpportunity). A list of SkillOpportunity
+            domain objects.
+    """
+    skill_opportunity_models = []
+    for skill_opportunity in skill_opportunities:
+        skill_opportunity.validate()
+        model = opportunity_models.SkillOpportunityModel(
+            id=skill_opportunity.id,
+            skill_description=skill_opportunity.skill_description,
+            question_count=skill_opportunity.question_count,
+        )
+        skill_opportunity_models.append(model)
+    opportunity_models.SkillOpportunityModel.put_multi(skill_opportunity_models)
+
+
+def update_skill_opportunity_skill_description(skill_id, new_description):
+    """Updates the skill_description of the SkillOpportunityModel with
+    new_description.
+
+    Args:
+        skill_id: str. The corresponding skill_id of the opportunity.
+        new_description: str. The new skill_description.
+    """
+    skill_opportunity = _get_skill_opportunity(skill_id)
+    if skill_opportunity:
+        skill_opportunity.skill_description = new_description
+        _save_skill_opportunities([skill_opportunity])
+
+
+def _get_skill_opportunity(skill_id):
+    """Returns the SkillOpportunity domain object representing a
+    SkillOpportunityModel with the supplied skill_id in the datastore.
+
+    Args:
+        skill_id: str. The corresponding skill_id of the opportunity.
+
+    Returns:
+        SkillOpportunity or None. The domain object representing a
+            SkillOpportunity with the supplied skill_id, or None if it does not
+            exist.
+    """
+    skill_opportunity_model = (
+        opportunity_models.SkillOpportunityModel.get_by_id(skill_id))
+    if skill_opportunity_model:
+        return get_skill_opportunity_from_model(skill_opportunity_model)
+    return None
+
+
+def delete_skill_opportunity(skill_id):
+    """Deletes the SkillOpportunityModel corresponding to the supplied skill_id.
+
+    Args:
+        skill_id: str. The skill_id corresponding to the to-be-deleted
+            SkillOpportunityModel.
+    """
+    skill_opportunity_model = (
+        opportunity_models.SkillOpportunityModel.get_by_id(skill_id))
+    opportunity_models.SkillOpportunityModel.delete(skill_opportunity_model)
+
+
+def update_skill_opportunity_question_counts(skill_ids, delta):
+    """Increments question_count(s) of SkillOpportunityModel(s) with
+    corresponding skill_ids.
+
+    Args:
+        skill_ids: list(str). A list of skill_ids corresponding to
+            SkillOpportunityModel(s).
+        delta: int. The delta for which to update each question_count.
+    """
+    updated_skill_opportunities = _update_skill_opportunity_question_count(
+        skill_ids, delta)
+    _save_skill_opportunities(updated_skill_opportunities)
+
+
+def update_question_count_by_skill_ids(old_skill_ids, new_skill_ids):
+    """Updates question_count(s) of SkillOpportunityModel(s) by the set
```

Comment:
This docstring is also quite confusing. You're updating the question counts (which are numbers) but that means it doesn't make sense to say that you're updating it by the "set difference of new_skill_ids - old_skill_ids" (which is a set of strings).